### PR TITLE
Clean docs and add internal documentation

### PR DESCRIPTION
**Pull request by elenatorro**
I added internal info about the new approach to release and modify documentation. In addition, I removed the `old-guides` directory which is not needed anymore and makes now a bit of noise, along with a couple of other old guides

### CODE REVIEW

File: docs/developers/documentation.rst [hunk 1]
```diff
@@ -68,3 +71,53 @@ Add documentation to the API reference:
     :members:
 
 - Include the file you want to import in the `__init__.py` file properly
+
+Developer center
```
Comment (Jesus89) [inline docs/developers/documentation.rst]: :heart: docs about docs

File: docs/developers/documentation.rst [hunk 2]
```diff
@@ -68,3 +71,53 @@ Add documentation to the API reference:
...
+
+- In order to test it in the developer center, it's necessary to change the branch to be tested in the `config.js` (instructions are explained in the developer center repository)
+- In order to modify the documentation of a specific version, just:
+    
+Download the docs branch and create from it a new one:
+
+.. code::
+    $ git fetch origin docs/v.M.m.p
+    $ git checkout docs/v.M.m.p
+    $ git checkout -b docs/v.M.m.p-your-fix-description
```
Comment (Jesus89) [inline docs/developers/documentation.rst]: We should try to avoid creating PRs with the `docs/*` because we can not remove these branches.

Comment (elenatorro) [reply docs/developers/documentation.rst]: OUCH!! veeeery good catch!! let me change that

File: docs/developers/documentation.rst [hunk 3]
```diff
@@ -68,3 +71,56 @@ Add documentation to the API reference:
...
+    $ git checkout -b your-fix-description
+
+Apply the necessary changes:
+
+    Note: If you're changing the examples and want to test them locally, don't forget to use the cartoframes version you've to use for the examples.
+
+.. code::
+    $ git add .
+    $ git commit -m "Commit description"
+    $ git push origin docs/v.M.m.p-your-fix-description
```
Comment (Jesus89) [inline docs/developers/documentation.rst]: Same here

**Pull request merged**